data/data/bootstrap/bootstrap-in-place/files/opt/openshift/bootstrap-in-place/bootstrap-in-place-post-reboot.sh

```diff
@@ -40,26 +43,31 @@ function restart_kubelet {
     echo "Waiting for kube-apiserver to apply the new static pod configuration"
     sleep 10
   done
+
+  echo "Restarting kubelet"
   systemctl restart kubelet
 }
 
 function approve_csr {
-  echo "Approving csrs ..."
+  echo "Waiting for node to report ready status"
   # use [*] and not [0] in the jsonpath because the node resource may not have been created yet
   until [ "$(oc get nodes --selector='node-role.kubernetes.io/master' -o jsonpath='{.items[*].status.conditions[?(@.type=="Ready")].status}' | grep -c "True")" -eq 1 ];
   do
     echo "Approving csrs ..."
     oc get csr -o go-template='{{range .items}}{{if not .status}}{{.metadata.name}}{{"\n"}}{{end}}{{end}}' | xargs --no-run-if-empty oc adm certificate approve &> /dev/null || true
-    sleep 30
+    sleep 10
   done
+  echo "node is ready"
 }
 
 function wait_for_cvo {
   echo "Waiting for cvo"
-  until [ "$(oc get clusterversion -o jsonpath='{.items[0].status.conditions[?(@.type=="Available")].status}')" == "True" ];
+  until [ "$(oc get clusterversion -o jsonpath='{.items[*].status.conditions[?(@.type=="Available")].status}')" == "True" ];
   do
     echo "Still waiting for cvo ..."
-    sleep 30
+    # print the not ready operators names and message
+    oc get clusteroperator -o jsonpath='{range .items[*]}{@.metadata.name}: {range @.status.conditions[?(@.type=="Available")]}{@.type}={@.status} {@.message}{"\n"}{end}{end}' | grep "Available=False" || true
```

Comment:
nit: because the CVO is waiting for (among other things) `Available=True`, you might want `grep -v Available=True` here.  That way any stray `Available=Unknown`, etc. would get included in the output too.